File: 15-18/ZXingDemo/app/src/main/java/com/wx/zxingdemo/MainActivity.java
package com.wx.zxingdemo;

import android.databinding.ViewDataBinding;
import android.view.View;

import com.wx.zxingdemo.activity.CreateQrActivity;
import com.wx.zxingdemo.activity.ScanActivity;
import com.wx.zxingdemo.base.BaseActivity;
import com.wx.zxingdemo.databinding.ActivityMainBinding;

//项目中要使用，写个demo熟悉一下
public class MainActivity extends BaseActivity implements View.OnClickListener{

    private ActivityMainBinding binding;

    @Override
    public int getLayout() {
        return R.layout.activity_main;
    }

    @Override
    public void setDataBinding(ViewDataBinding binding) {
        this.binding = (ActivityMainBinding) binding;
    }

    @Override
    public void initData() {
        binding.btnScan.setOnClickListener(this);
        binding.btnGenerate.setOnClickListener(this);
    }

    @Override
    public void onClick(View view) {
        switch (view.getId()) {
            case R.id.btn_generate:
                openActivity(CreateQrActivity.class);
                break;
            case R.id.btn_scan:
                openActivity(ScanActivity.class);
                break;
        }
    }
}

File: Android-Demo/WebView/app/src/main/java/com/wx/webview/MainActivity.kt
package com.wx.webview

import android.databinding.ViewDataBinding
import com.wx.glidedemo1.base.BaseActivity
import com.wx.webview.activity.BasicUsageActivity
import com.wx.webview.activity.JsBridgeActivity
import com.wx.webview.databinding.ActivityMainBinding

class MainActivity : BaseActivity() {

    private lateinit var binding: ActivityMainBinding

    override val layoutId: Int
        get() = R.layout.activity_main

    override fun setDataBinding(binding: ViewDataBinding) {
        this.binding = binding as ActivityMainBinding
    }

    override fun initData() {
        binding.btnBasicUsage.setOnClickListener {
            openActivity(BasicUsageActivity::class.java)
        }
        binding.btnJsBridge.setOnClickListener {
            openActivity(JsBridgeActivity::class.java)
        }
    }


}
